### ISSUE TITLE
MW Advanced Search Display Suggestion

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/COLOGNE#185]
Odile, a longtime user of MW and the French dictionaries, made this comment involving advanced search
display of MW:
```
This short email to inform you of  some difficulty i find on the advanced research when 
looking for with  prefix or suffix research. 
You recently changed the format of the list on the left, and i very much regret the 
old presentation. 
Why? Because it is now difficult to find quickly a word in the list as the is a blank line 
between each word.
Also, the previous font was more legible.
```

**Comment by funderburkjim** [comment]
Here's my response to Odile:
```
 You are right that we recently changed the font for Devanagari in the MW displays.  We changed to
  use the 'Siddhanta' font  (see http://svayambhava.blogspot.com/p/siddhanta-devanagariunicode-open-type.html).  The reason was the ligatures;  such as 'kt'; several users stated a preference for the
way Siddhanta renders ligatures in comparison to the  default Windows font for Devanagari
characters (currently Nirmala font in Windows 10).

I currently have a local version of the display using the old font, and have made a screen shot of the
Advanced search word list side by side (see attachment).
The new Siddhanta font is on left, the old default font is on right.
The differences that I see are:
1.  There are fewer entries in the new font
2.  The spacing between the lines is greater in the new font
3.  The size of letters in the new font is smaller
4.  The new font appears slightly 'lighter' in color.

Are these the main differences that you notice?

It should be possible to alter some of the features of the new font so that these differences between
 the new and old fonts are lessened.  This kind of 'styling' adjustment involves minor changes to 
what is known as CSS (cascading style sheets) , and I'm not very skilled at it.
  But I'll give it a try in the next week or two.
```
![new_vs_old](https://user-images.githubusercontent.com/6393033/31198042-f504ed1a-a920-11e7-862d-d53068293388.PNG)

**Comment by funderburkjim** [comment]
From using the Google Chrome 'Inspect Element' feature,  it is found that the only details of difference
in the CSS parameters is that
the font is Siddhanta on left and Nirmala UI on the right.

The computed font size shows as identical (15.6 px) in both.

The line-spacing is accomplished by an html `<br/>`  break.

Yet there clearly is a difference in both line-spacing and font size in the two.  
1. Any idea how to understand the differences ?
2. Any idea what needs to be tweaked to get the printed size and line spacing of the Siddhanta list
more similar to the detaul list?

**Comment by gasyoun** [comment]
> Any idea how to understand the differences?

Scripts (instructions) inside the font - differ. Nothing we can do about it. But indeed - in this case the line spacing is too big. 

> Any idea what needs to be tweaked to get the printed size and line spacing of the Siddhanta list
more similar to the detail list?

```
	line-height: 12px;
	font-size: larger;
	font-family: Siddhanta,Nirmala;
```

12 might be too small, experiments needed. Exact size could fix the issue. Font size needs to be increased, not another way around.

**Comment by SergeA** [comment]
> Scripts (instructions) inside the font - differ.

To be precise, this is not about scripts, but glyph sizes. As Siddhanta is designed for the use with vedic accents, there are reserved spaces above and below the characters in it.

**Comment by funderburkjim** [comment]
###  space between lines also with IAST (Roman)

I was assuming Odile displayed output in Devanagari.  However, she further explained that she actually is
seeing the problem with Roman Unicode (IAST) output as well.  She adds
```
In addition, in Roman script, the new font is of a serif type, that is less legible on screen all the more 
so as yes it is thinner. For that aspect, you could try to use the bold set instead, 
but that will not be solving the main problem, which is the blank line.

The fact it is a bit thinner can be tolerate, but not the blank line.
```

As following image shows, there is the same spacing difference here also:
![image](https://user-images.githubusercontent.com/6393033/31249716-59d96dea-a9e6-11e7-873a-ea9bc3155e27.png)

From Dev. Tools, the rendered fonts show as:
* Siddhanta  (yes, still Siddhanta) for the left side (current Cologne version) -- with extra spacing
* Arial  --  for the other.

> there are reserved spaces above

This may account for both for the smaller glyph size AND the extra space between lines.

#### Specified font-family vs. rendered font
This information from Chrome Dev Tools 'Inspect Element' for a word in left-hand list.

* Local version:  css font-family = `arial,helvetica,verdana,sansserif;` .
  * Devanagari output, rendered fonts = Nirmala
  * IAST output, rendered fonts = Arial
* Cologne version: css font-family = font-family: siddhanta_deva,arial,helvetica,verdana,sansserif;
  * Devanagari output, rendered fonts = Siddhanta—Network resource(7 glyphs)
  * IAST output, rendered fonts = Siddhanta—Network resource(7 glyphs)

Note how the rendered font changes for the local version, but not for the Cologne version.
In other words, it appears that Siddhanta 'takes over' the IAST font rendering, instead of passing it
over to the next font (arial).   I would have expected Siddhanta to apply *only* to characters in the Unicode code blocks for Devanagari, but apparently this is a wrong assumption.

**Comment by funderburkjim** [comment]
### line-height 

This proves to be the right parameter to control the spacing between lines.
After some fiddling,  it turns out that `line-height=18px`  gives spacing on Cologne-Siddhanta version
same as prior version for Roman Unicode output:
![image](https://user-images.githubusercontent.com/6393033/31251890-6ed2c910-a9ed-11e7-9fe7-6cdfa3504b2d.png)

The comparison is similar for Devanagari -- though the Siddhanta still appears smaller (due to smaller glyphs ),  the spacing seems quite acceptable.

This line-spacing change seems to solve the main problem Odile mentioned.
Thanks, @gasyoun 👍

**Comment by funderburkjim** [comment]
#### changing font-family order 

A change in the order of the css font-family list in Cologne display: 
```
font-family: siddhanta_deva,arial,helvetica,verdana,sansserif; /* 08-21-2017 PREVIOUS */
font-family: arial,siddhanta_deva,helvetica,verdana,sansserif; /* 10-05-2017 CURRENT */
```
This has an effect on the rendered font:
* For Roman Unicode output , Arial is used
* For Devanagari output, siddhanta is used.

![image](https://user-images.githubusercontent.com/6393033/31252331-dee591aa-a9ee-11e7-949b-37ab34d9cb7b.png)

**Comment by gasyoun** [comment]
You've become a CSS 2 master by now, Jim. I'm no fan of Arial, but let it be. The change applies to only 1 dictionary, that's the bad part about it.

**Comment by funderburkjim** [comment]
>applies to only 1  dictionary.

When the meta-line conversion ordeal is done (or at least done for the main dictionaries), the next infrastructure improvement is to put the various displays on one code base.  What this will mean is that when we think of an improvement, we can apply it to all the (main) dictionaries.
